services/codeship/codeship.service.js

```diff
@@ -1,19 +1,44 @@
 'use strict'
 
-const LegacyService = require('../legacy-service')
-const { makeBadgeData: getBadgeData } = require('../../lib/badge-data')
+const Joi = require('joi')
+const { BaseSvgScrapingService } = require('..')
+const { isBuildStatus, renderBuildStatusBadge } = require('../build-status')
 
-// This legacy service should be rewritten to use e.g. BaseJsonService.
-//
-// Tips for rewriting:
-// https://github.com/badges/shields/blob/master/doc/rewriting-services.md
-//
-// Do not base new services on this code.
-module.exports = class Codeship extends LegacyService {
+const schema = Joi.object({
+  message: Joi.alternatives()
+    .try(
+      isBuildStatus,
+      Joi.equal('project not found', 'branch not found', 'ignored', 'blocked')
+    )
+    .required(),
+}).required()
+
+const pendingStatus = 'pending'
+const notBuiltStatus = 'not built'
+
+const statusMap = {
+  testing: pendingStatus,
+  waiting: pendingStatus,
+  initiated: pendingStatus,
+  stopped: notBuiltStatus,
+  ignored: notBuiltStatus,
+  blocked: notBuiltStatus,
+}
+
+module.exports = class Codeship extends BaseSvgScrapingService {
```

Comment:
There is a JSON based API. However, it's rate limited (60 requests per minute) so I figured it would make sense to continue using the SVG.

Reply:
Agreed. The other alternative would be to see if they'd provide us a mega-token with a higher quota.

Reply:
👍 Once we starting get a count of how frequently this is badge is used we'll have a better idea. If it's only a few per minute then that rate limit may not be even an issue